cmake/modules/platform/toolcfg/gnu.cmake

```diff
@@ -35,6 +35,10 @@ list(APPEND OMR_PLATFORM_COMPILE_OPTIONS -pthread -fno-strict-aliasing)
 
 list(APPEND OMR_PLATFORM_CXX_COMPILE_OPTIONS -std=c++0x)
 
+if(CODE_COVERAGE)
+	list(APPEND OMR_PLATFORM_COMPILE_OPTIONS -fprofile-arcs -ftest-coverage)
```

Comment:
What about other tool chains?
Where is the definition of `CODE_COVERAGE` expected to originate?
Would something more general be appropriate (e.g. `OMR_EXTRA_CFLAGS` and `OMR_EXTRA_CXXFLAGS`)?

Reply:
Thanks for reviewing!
- Currently we are enabling code coverage for xlinux, zlinux, and plinux first, and then expand to other platforms later.
- The `CODE_COVERAGE` setting is in another PR https://github.com/eclipse-openj9/openj9/pull/13386, where if user checks `param.CODE_COVERAGE` in build pipeline, we will set `EXTRA_CMAKE_ARGS=-DCODE_COVERAGE=ON` to enable it.
- For more general `EXTRA_FLAGs`, do you mean I can put `-fprofile-arcs -ftest-coverage` into an extra variable (e.g. `OMR_EXTRA_PLATFORM_COMPILE_FLAGS`), and then append this variable to `OMR_PLATFORM_COMPILE_OPTIONS` if `CODE_COVERAGE` is ON? Thanks.

Reply:
This change will also apply to Linux on aarch64 and risc-v: Did you verify that the additional options are supported for those architectures?

After asking my question, I found eclipse-openj9/openj9#13386: Now I agree that specifying `CODE_COVERAGE` is preferable to adding tool-specific options to `OMR_EXTRA_*_FLAGS`.

Reply:
Comments in https://github.com/eclipse-openj9/openj9/pull/13386 have been addressed.
New changes also passed aarch64 (Internal_Jenkins/job/Build_JDK11_aarch64_linux_Personal/176/), while risc-v seems not available in `Pipeline-Build-Test-Personal`. However, since we have a checkbox for `params.CODE_COVERAGE` for pipeline, we can only enabled this parameter for xlinux, plinux, zlinux, and aarch, and not select this checkbox for other platforms. Thanks.